misago/users/signals.py

```diff
@@ -9,3 +20,14 @@
 @receiver(username_changed)
 def handle_name_change(sender, **kwargs):
     sender.user_renames.update(changed_by_username=sender.username)
+
+
+@receiver(anonymize_old_ips)
+def anonymize_old_registrations_ips(sender, **kwargs):
+    anonymization_cutoff = timezone.now() - timedelta(days=settings.MISAGO_IP_STORE_TIME)
+    ip_is_too_new = Q(joined_on__gt=anonymization_cutoff)
+    ip_is_already_anonymized = Q(joined_from_ip=ANONYMOUS_IP)
+
+    queryset = UserModel.objects.exclude(ip_is_too_new | ip_is_already_anonymized)
+    queryset.update(joined_from_ip=ANONYMOUS_IP)
+
```

Comment:
So far its looking good. I would now like for you to create additional receiver function named `anonymize_old_last_used_ips` that will anonymize user's IP address stored on the `last_ip` field if time since user's last visit was too far in past.